src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue

```diff
@@ -0,0 +1,291 @@
+  props: {
+    props: {
+      type: Object as () => EceWorkExperienceReferencesProps,
+      required: true,
+    },
+    modelValue: {
+      type: Object as () => { [id: string]: Components.Schemas.WorkExperienceReference },
+      required: true,
+    },
+  },
+  emits: {
+    "update:model-value": (_referenceData: { [id: string]: Components.Schemas.WorkExperienceReference }) => true,
+  },
+  setup: () => {
+    const alertStore = useAlertStore();
+
+    return {
+      alertStore,
+    };
+  },
+  data: function () {
+    return {
+      clientId: "",
+      mode: "add",
+      id: null as string | null,
+      previousFullName: "",
+      firstName: "",
+      lastName: "",
+      email: "",
+      phoneNumber: "",
+      hours: "",
+      Rules,
+    };
+  },
+  computed: {
+    isDisabled() {
+      return this.count >= 6 || this.totalHours >= 500;
+    },
+    totalHours() {
+      return Object.values(this.modelValue).reduce((acc, reference) => {
+        return acc + parseInt(reference.hours as string);
+      }, 0);
+    },
+    count() {
+      return Object.keys(this.modelValue).length;
+    },
+    fullName() {
+      return `${this.firstName} ${this.lastName}`;
+    },
+    newClientId() {
+      return Object.keys(this.modelValue).length + 1;
+    },
+  },
+  mounted() {
+    if (Object.keys(this.modelValue).length === 0) {
+      this.mode = "add";
+    } else {
+      this.mode = "list";
+    }
+  },
+  methods: {
+    isNumber,
+    async handleSubmit() {
+      // Validate the form
+      const { valid } = await (this.$refs.addWorkExperienceReferenceForm as any).validate();
+
+      if (valid) {
+        // Prepare the new or updated WorkExperienceReference data
+        const newReference: Components.Schemas.WorkExperienceReference = {
+          id: this.id,
+          firstName: this.firstName,
+          lastName: this.lastName,
+          emailAddress: this.email,
+          phoneNumber: this.phoneNumber,
+          hours: this.hours,
+        };
+
+        // see if we already have a clientId (which is edit), if not use the newClientId (which is add)
+        const clientId = this.clientId ? this.clientId : this.newClientId;
+        // Update the modelValue dictionary
+        const updatedModelValue = {
+          ...this.modelValue,
+          [clientId]: newReference,
+        };
+
+        // Emit the updated modelValue
+        this.$emit("update:model-value", updatedModelValue);
+
+        // Set success alert message
+        const message = this.modelValue[clientId] ? "You have successfully edited your Reference." : "You have successfully added your Reference.";
+        this.alertStore.setSuccessAlert(message);
+
+        // Change mode to list
+        this.mode = "list";
+      } else {
+        this.alertStore.setFailureAlert("Please fill out all required fields");
+      }
+    },
+    handleCancel() {
+      // Change mode to list
+      this.mode = "list";
```

Comment:
Do we need to also resetFormData here? 

I suspect if we try to add/edit a user. Cancel it and then try to add a new user. Will it retain the info we input previously?

Reply:
No we're OK, we reset before add/edit.